client/tink_keyset_helper.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/daead"
+	"github.com/google/tink/go/hybrid"
+	"github.com/google/tink/go/mac"
+	"github.com/google/tink/go/signature"
+	"github.com/google/tink/go/streamingaead"
+
+	tinkpb "github.com/google/tink/go/proto/tink_go_proto"
+)
+
+// Represents the info for a supported tink keyset template.
+type tinkKeyTemplateInfo struct {
+	knoxIDPrefix string
+	templateFunc func() *tinkpb.KeyTemplate
+}
+
+// This map contains the supported tink key templates and the correcsponding naming rule for knox identifier
+var tinkKeyTemplates = map[string]tinkKeyTemplateInfo{
+	"TINK_AEAD_AES256_GCM":                               {"tink:aead:", aead.AES256GCMKeyTemplate},
```

Comment:
are these tink template names defined in the tink library? It might be better to use their strings / consts than define our own strings (could have typos / etc)

Reply:
The key template constants are not defined in the golang version of tink.  They are defined in Java version as it's used as the base for [tinkey](https://developers.google.com/tink/tinkey-overview) which is a key management utility. Hence, we are hardcoding the key template constants here.

Reply:
Tink library doesn't have a const name, there is a const TypeURL for each template function, I paste a example [here](https://github.com/google/tink/blob/7c93a224b8fa6a3babfaf71c18c5610052dcbd61/go/aead/aes_gcm_key_manager.go#L33). However, the name of TypeURL is not straight forward.

Also, the prefix of our own defined name contains two information: 1. this template is for Tink keyset, also, which type key (aead, or dsig) 2. the knox identifier needs to have same prefix in lower case.
I paste a wiki page that I wrote [here](https://w.pinadmin.com/display/SEC/Tink+Keyset+Management+With+Knox), which have more details.